lib/ramble/ramble/reports.py

```diff
@@ -0,0 +1,989 @@
+        self.output_df = pd.DataFrame()
+
+        self.logx = logx
+        self.logy = logy
+
+        self.split_by = split_by
+
+        # TODO: enable more robustly
+        self.have_statistics = False
+        self.better_direction = BetterDirection.INDETERMINATE
+
+    def normalize_data(
+        self,
+        data,
+        scale_to_index=False,
+        to_col=ReportVars.NORMALIZED_FOM_VALUE.value,
+        from_col=ReportVars.FOM_VALUE.value,
+        speedup=False,
+    ):
+        # FIXME: do we need to support more than normalizing by the first
+        # value?
+        if data[from_col].iloc[0] == 0:
+            raise ArithmeticError(
+                "Unable to normalize data. The first value in the series cannot be zero."
+            )
+        else:
+            # Adjusts first y-value to first scale var when >1 (e.g., speedup for 2+ nodes = 2)
+            if scale_to_index:
+                # Performs inplace edit on data, no need to return
+                if speedup:
+                    data.loc[:, to_col] = (
+                        data[from_col].iloc[0] / data.loc[:, from_col]
+                    ) * data.index[0]
+                else:
+                    data.loc[:, to_col] = (
+                        data.loc[:, from_col] / data[from_col].iloc[0]
+                    ) * data.index[0]
+            else:
+                if speedup:
+                    data.loc[:, to_col] = data[from_col].iloc[0] / data.loc[:, from_col]
+                else:
+                    data.loc[:, to_col] = data.loc[:, from_col] / data[from_col].iloc[0]
+
+    def add_minmax_data(self, selected_data, min_data, max_data, scale_var):
+        """When using summary statistics from repeats, adds columns fom_value_min and fom_value_max
+        to the selected data.
+        """
+        min_data.loc[:, scale_var] = to_numeric_if_possible(min_data[scale_var])
+        min_data = min_data.set_index(scale_var)
+        max_data.loc[:, scale_var] = to_numeric_if_possible(max_data[scale_var])
+        max_data = max_data.set_index(scale_var)
+
+        selected_data.loc[:, ReportVars.FOM_VALUE_MIN.value] = to_numeric_if_possible(
+            min_data[ReportVars.FOM_VALUE.value]
+        )
+        selected_data.loc[:, ReportVars.FOM_VALUE_MAX.value] = to_numeric_if_possible(
+            max_data[ReportVars.FOM_VALUE.value]
+        )
+
+        if self.normalize:
+            self.normalize_data(
+                selected_data,
+                scale_to_index=True,
+                to_col=ReportVars.FOM_VALUE_MIN.value,
+                from_col=ReportVars.FOM_VALUE_MIN.value,
+            )
+            self.normalize_data(
+                selected_data,
+                scale_to_index=True,
+                to_col=ReportVars.FOM_VALUE_MAX.value,
+                from_col=ReportVars.FOM_VALUE_MAX.value,
+            )
+
+    # TODO: these args come from the spec, so don't need to be passed and could be stored at init
```

Comment:
Is this easy to do?

Reply:
I think this is actually no longer true, will remove